### PR TITLE
Fix RunsWithDatanode injection for insecure datanodes instances

### PR DESCRIPTION
**Pull request by todvora**
/nocl

## Description
The preflight finished flag will be never configured if the datanode is configured in insecure mode and a graylog server starts against such datanode. This then blocks other features depending on the datanode detection.

To discussion:

- What happens if a datanode appears but it's not used as indexer?
- Is the caching there making the functionality better or worse?
- What happens if the datanode server starts after the graylog server?

The only really reliable solution would be to extend VersionProbe in a way that guarantees that we are using the datanode as the indexer. But there is nothing, except changeable cluster name, that we could use to detect this. Maybe matching the indexer url with datanodes collection in mongodb and know that if the url matches and we get back opensearch version, it's actually a datanode?

## Motivation and Context
Fixes https://github.com/Graylog2/graylog-plugin-enterprise/issues/6835

## How Has This Been Tested?
Existing set of integration tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/configuration/RunsWithDataNodeDiscoveryProvider.java [hunk 1]
```diff
@@ -57,24 +44,6 @@ public Boolean get() {
...
-        LOG.debug("Not running with DataNode(s).");
-        return Boolean.FALSE;
-    }
-
-    private List<URI> discover() {
-        return nodeService.allActive().values().stream()
-                .map(Node::getTransportAddress)
-                .map(URI::create)
-                .collect(Collectors.toList());
+        return !nodeService.allActive().isEmpty();
```
Comment (moesterheld) [inline graylog2-server/src/main/java/org/graylog2/configuration/RunsWithDataNodeDiscoveryProvider.java:47]: would it make sense to add an additional filter for `DataNodeStatus.AVAILABLE` here?

**Review by moesterheld — CHANGES_REQUESTED** [review]
lgtm, but maybe we could improve it by adding the mentioned check?

**Comment by bernd** [comment]
> The only really reliable solution would be to extend VersionProbe in a way that guarantees that we are using the datanode as the indexer. But there is nothing, except changeable cluster name, that we could use to detect this. Maybe matching the indexer url with datanodes collection in mongodb and know that if the url matches and we get back opensearch version, it's actually a datanode?

@todvora How about setting a custom `node.attr` value for our Data Node managed OpenSearch nodes? We can query the node attributes with the `/_cat/nodeattrs` API.

**Comment by todvora** [comment]
Thanks @bernd. This could be very useful and we should look into this. Maybe also setting the exact datanode version here.

**Comment by janheise** [comment]
I disagree - I was happy with `Preflight.FINISHED` and would have added/renamed it. Now, with both solutions (in the PR, and Bernd's suggested solution) it's tied to the runtime state of OpenSearch. And that is not the same as knowing that it's the DataNode, even without a running OpenSearch.

**Comment by bernd** [comment]
> I disagree - I was happy with `Preflight.FINISHED` and would have added to the insecure startup and maybe renamed it to a better name. Now, with both solutions (in the PR, and Bernd's suggested solution) it's tied to the runtime state of OpenSearch. And that is not the same as knowing that it's the DataNode, even without a running OpenSearch.

Good point! :+1:

**Comment by todvora** [comment]
@janheise IDK how to fix this otherwise. If you have any suggestion please feel free to refactor or replace this PR with another one :pray:

**Comment by janheise** [comment]
@todvora I'll see that I come up with a PR for my suggested solution above. Maybe it can get replaced by sth. better for the first bugfix release.

**Comment by janheise** [comment]
@todvora see https://github.com/Graylog2/graylog2-server/pull/18796

**Comment by janheise** [comment]
we implemented/merged https://github.com/Graylog2/graylog2-server/pull/18796, so I'm closing this PR.

**Pull request closed without merging**